assets/prototype/application/ui/components/layout/eeditorModal/useEditorModalManager.ts

```diff
@@ -0,0 +1,80 @@
+import { useReducer, useEffect } from 'react';
+
+import {
+	EditorState,
+	EditorId,
+	EditorStateManager,
+	EditorStateReducer,
+	ActionType,
+	OpenEditorOptions,
+	EntityId,
+} from './types';
+
+export const INITIAL_STATE: EditorState = {
+	isOpen: false,
+	editorId: null,
+	entityId: '',
+};
+
+const useEditorModalManager = (): EditorStateManager => {
+	const [state, dispatch] = useReducer<EditorStateReducer>(reducer, INITIAL_STATE);
+
+	useEffect(() => {
+		console.log('<<<EditorModalManager>>>', state);
+	}, [state]);
+
+	const openEditor = (options: OpenEditorOptions): void => {
+		dispatch({
+			...options,
+			type: ActionType.OPEN_EDITOR,
+			isOpen: true,
+		});
+	};
+
+	const setEditorId = (editorId: EditorId): void => {
+		dispatch({
+			type: ActionType.SET_EDITOR_ID,
+			editorId,
+		});
+	};
+
+	const setEntityId = (entityId: EntityId): void => {
+		dispatch({
+			type: ActionType.SET_ENTITY_ID,
+			entityId,
+		});
+	};
```

Comment:
I am not sure I understand the purpose of this.
My understanding was that when `useEditorModalManager` is executed, then we pass the entity id.
Please help me understand this piece.